game.py

```diff
@@ -20,7 +20,28 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    correct_letter_guess_statuses = build_letter_status_dict(snowman_word)
+    wrong_guesses_list = []
+    while len(wrong_guesses_list) < SNOWMAN_MAX_WRONG_GUESSES and not (is_word_guessed(snowman_word, correct_letter_guess_statuses)):
+        user_input = get_letter_from_user(correct_letter_guess_statuses, wrong_guesses_list)
+        if user_input in correct_letter_guess_statuses:
+            correct_letter_guess_statuses[user_input] = True 
+            print("You guessed a letter that's in the word!")
+        else:
+            print(f"The letter {user_input} is not in the word!")
+            wrong_guesses_list.append(user_input)
+        wrong_guesses_count = len(wrong_guesses_list)
+        print_snowman_graphic(wrong_guesses_count)
+        
+        output_string = generate_word_progress_string(snowman_word, correct_letter_guess_statuses)
+        print_word_progress_string(snowman_word, correct_letter_guess_statuses)
+        print(f"Wrong guesses: {wrong_guesses_list}")
+    if is_word_guessed(snowman_word, correct_letter_guess_statuses):
+        print("Congratulations, you win!")
+        return 'Congratulations, you win!'
```

Comment:
We typically want to avoid duplicating text and information since it also introduces an opportunity for a bug to occur. This commonly happens when the data is modified in multiple places, but we don't remember to update our duplicate data in every location.

Another option here would be to create a variable that holds the text, then print and return that variable.